fv3core/stencils/fxadv.py

```diff
@@ -1,314 +1,381 @@
-        grid().rsin_v,
-        grid().sin_sg2,
-        grid().sin_sg4,
-        vt_in,
-    )
-    update_ut_y_edge(uc_in, grid().sin_sg1, grid().sin_sg3, ut, dt)
-    update_vt_y_edge(vc_in, grid().cosa_v, ut, vt)
-    update_vt_x_edge(vc_in, grid().sin_sg2, grid().sin_sg4, vt, dt)
-    update_ut_x_edge(uc_in, grid().cosa_u, vt, ut)
-    corner_shape = (1, 1, uc_in.shape[2])
-    if grid().sw_corner:
-        sw_corner(uc_in, vc_in, ut, vt, grid().cosa_u, grid().cosa_v, corner_shape)
-    if grid().se_corner:
-        se_corner(uc_in, vc_in, ut, vt, grid().cosa_u, grid().cosa_v, corner_shape)
-    if grid().ne_corner:
-        ne_corner(uc_in, vc_in, ut, vt, grid().cosa_u, grid().cosa_v, corner_shape)
-    if grid().nw_corner:
-        nw_corner(uc_in, vc_in, ut, vt, grid().cosa_u, grid().cosa_v, corner_shape)
-    ra_x = utils.make_storage_from_shape(
-        uc_in.shape, grid().compute_origin(add=(0, -grid().halo, 0))
-    )
-    xfx_adv_stencil(
-        ut,
-        grid().rdxa,
-        grid().area,
-        grid().dy,
-        grid().sin_sg1,
-        grid().sin_sg3,
-        crx_adv,
-        xfx_adv,
-        ra_x,
-        dt,
-        origin=grid().compute_origin(add=(0, -grid().halo, 0)),
-        domain=grid().domain_shape_compute(add=(1, 2 * grid().halo, 0)),
-    )
-    ra_y = utils.make_storage_from_shape(
-        vc_in.shape, grid().compute_origin(add=(-grid().halo, 0, 0))
-    )
-    yfx_adv_stencil(
-        vt,
-        grid().rdya,
-        grid().area,
-        grid().dx,
-        grid().sin_sg2,
-        grid().sin_sg4,
-        cry_adv,
-        yfx_adv,
-        ra_y,
-        dt,
-        origin=grid().compute_origin(add=(-grid().halo, 0, 0)),
-        domain=grid().domain_shape_compute(add=(2 * grid().halo, 1, 0)),
-    )
-    # TODO: Remove the need for a copied extra ut and vt variables, edit in place
-    # (rexolve issue with data getting zeroed out).
-    ut_in[:, :, :] = ut[:, :, :]
-    vt_in[:, :, :] = vt[:, :, :]
-    return ra_x, ra_y
-
-
-def compute_ut(uc_in, vc_in, cosa_u, rsin_u, ut_in):
-    ut_origin = (grid().is_ - 1, grid().jsd, 0)
-    ut = utils.make_storage_from_shape(ut_in.shape, ut_origin)
-    main_ut(
-        uc_in,
-        vc_in,
-        cosa_u,
-        rsin_u,
-        ut,
-        origin=ut_origin,
-        domain=(grid().nic + 3, grid().njd, grid().npz),
-    )
-    ut[: grid().is_ - 1, :, :] = ut_in[: grid().is_ - 1, :, :]
-    ut[grid().ie + 3 :, :, :] = ut_in[grid().ie + 3 :, :, :]
-    # fill in for j /=2 and j/=3
-    if grid().south_edge:
-        ut[:, grid().js - 1 : grid().js + 1, :] = ut_in[
-            :, grid().js - 1 : grid().js + 1, :
-        ]
-    # fill in for j/=npy-1 and j /= npy
-    if grid().north_edge:
-        ut[:, grid().je : grid().je + 2, :] = ut_in[:, grid().je : grid().je + 2, :]
-    return ut
 
+@gtscript.function
+def ut_corners(uc, vc, cosa_u, cosa_v, ut, vt):
+    from __externals__ import i_end, i_start, j_end, j_start
+
+    """
```

Comment:
I believe the convention is that the docstring is the first thing in the function.

Reply:
ok, will change.